bundles/org.openhab.binding.sncf/src/main/java/org/openhab/binding/sncf/internal/handler/SncfBridgeHandler.java

```diff
@@ -0,0 +1,177 @@
+            throw new SncfException("Unable to deserialize API answer");
+        }
+        if (response.message != null) {
+            throw new SncfException(response.message);
+        }
+        return response;
+    }
+
+    private String getResponse(String url) throws SncfException {
+        try {
+            logger.debug("SNCF Api request: URL = '{}'", url);
+            ContentResponse contentResponse = httpClient.newRequest(url).method(GET).timeout(10, TimeUnit.SECONDS)
+                    .header("Authorization", apiId).send();
+            int httpStatus = contentResponse.getStatus();
+            String content = contentResponse.getContentAsString();
+            logger.debug("SNCF Api response: status = {}, content = '{}'", httpStatus, content);
+            switch (httpStatus) {
+                case OK_200:
+                    return content;
+                case BAD_REQUEST_400:
+                case UNAUTHORIZED_401:
+                case NOT_FOUND_404:
+                    logger.debug("SNCF Api server responded with status code {}: {}", httpStatus, content);
+                    throw new SncfException(content);
+                default:
+                    logger.debug("SNCF Api server responded with status code {}: {}", httpStatus, content);
+                    throw new SncfException(content);
+            }
+        } catch (TimeoutException | ExecutionException e) {
+            logger.debug("Exception occurred during execution: {}", e.getLocalizedMessage(), e);
+            throw new SncfException(e.getLocalizedMessage(), e.getCause());
+        } catch (InterruptedException e) {
+            logger.debug("Execution interrupted: {}", e.getLocalizedMessage(), e);
+            Thread.currentThread().interrupt();
+            throw new SncfException(e.getLocalizedMessage(), e.getCause());
+        }
+    }
+
+    public @Nullable List<PlaceNearby> discoverNearby(PointType location, int distance) throws SncfException {
+        String URL = String.format(Locale.US,
+                "%s/coord/%.5f;%.5f/places_nearby?distance=%d&type[]=stop_point&count=100", SERVICE_URL,
+                location.getLongitude().floatValue(), location.getLatitude().floatValue(), distance);
+        PlacesNearby places = getResponseFromCache(URL, PlacesNearby.class);
+        return places.placesNearby;
+    }
+
+    public Optional<StopPoint> stopPointDetail(String stopPointId) throws SncfException {
+        String URL = String.format(Locale.US, "%s/stop_points/%s", SERVICE_URL, stopPointId);
+        List<StopPoint> points = getResponseFromCache(URL, StopPoints.class).stopPoints;
+        return points != null ? Optional.ofNullable(points.get(0)) : Optional.empty();
+    }
+
+    public Optional<Passage> getNextPassage(String stopPointId, String expected) throws SncfException {
+        String URL = String.format(Locale.US, "%s/stop_points/%s/%s?disable_geojson=true&count=1", SERVICE_URL,
+                stopPointId, expected);
```

Comment:
Is it necessary to pass a locale in this case ?

Reply:
You're correct, not needed.